ndc-client/src/models.rs

```diff
@@ -58,6 +59,16 @@ pub struct QueryCapabilities {
 }
 // ANCHOR_END: QueryCapabilities
 
+// ANCHOR: MutationCapabilities
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Mutation Capabilities")]
+pub struct MutationCapabilities {
+    /// Does the connector support executing multiple mutations in a transaction.
+    pub transactional: Option<LeafCapability>,
```

Comment:
I think we need to be clear about what this implies for the semantics of mutations, and I don't know that we can really do that. Perhaps we could do it with respect to reads (a mutation is transactional if its components can only be observed together as a unit, or not at all, where observed means by probing the data source using the query API).

But how will engine use this capability? The abstraction has to be useful from the consumer point of view as well, and I'm not seeing what this would enable or disable on the engine side.

Reply:
We discussed this on slack, the tl;dr is that this would enable the engine to know whether it should make a multiple requests with one mutation in each request, or a single request containing multiple mutations, when it receives a request with multiple mutations from the user.

The difference between the two is that with multiple mutations per request once we encounter an error we terminate and return the error and no data (and it seems that rolling back is the only reasonable semantics we can thinking of atm), and with single mutations per request we do not terminate, but return both the data and errors for their respective mutations.

How should we proceed with this PR? Do we want to reword these ideas and flags?

Reply:
Resolved previously - we will capture this in the specification with a note about error handling. It makes sense to call this "transactional", because we need to specify something along the lines of "all or none will be performed".